### PR TITLE
domain partitioning

### PR DESCRIPTION
**Pull request by MatildeMark**
Yihan Lu Domain partition code

### CODE REVIEW

**Comment by lucasccordeiro** [comment]
@MatildeMark: there are too many files in this PR. As far as I know, you have modified `bmc.cpp` and `bmc.h` and added two new files `bmc_domain_split.cpp` and `bmc_domain_split.h`. Is my interpretation correct?

File: src/solidity-frontend/solidity_convert.cpp [hunk 1]
```diff
@@ -215,21 +213,22 @@ bool solidity_convertert::get_var_decl(
...
-    assert(current_functionName != "");
-    get_var_decl_name(ast_node, name, id);
-  }
   else
   {
-    assert(
-      !"Error: ESBMC could not find the parent scope for this local variable");
+    if(!current_functionDecl)
+    {
+      assert(!"Error: ESBMC could not find the parent scope for this local variable");
```
Comment (kunjsong01) [inline src/solidity-frontend/solidity_convert.cpp:224]: The commit in this PR appears to have reverted lots of recent commits.

**Comment by mikhailramalho** [comment]
Maybe they are using Windows? There are a lot of file permission changes.

**Comment by lucasccordeiro** [comment]
Closed via https://github.com/esbmc/esbmc/pull/699.

**Pull request closed without merging**